sessionctx/variable/sysvar.go

```diff
@@ -2792,6 +2792,25 @@ var defaultSysVars = []*SysVar{
 	}, GetGlobal: func(ctx context.Context, vars *SessionVars) (string, error) {
 		return BoolToOnOff(EnableCheckConstraint.Load()), nil
 	}},
+	{Scope: ScopeSession, Name: TiDBSessionAlias, Value: "", Type: TypeStr,
+		Validation: func(s *SessionVars, normalizedValue string, originalValue string, _ ScopeFlag) (string, error) {
+			if len(originalValue) > 64 {
+				s.StmtCtx.AppendWarning(ErrTruncatedWrongValue.GenWithStackByArgs(TiDBServerMemoryLimitSessMinSize, originalValue))
+				normalizedValue = originalValue[:64]
+			}
+
+			if len(normalizedValue) > 0 && normalizedValue[len(normalizedValue)-1] == ' ' {
```

Comment:
IMO it's better to reuse [isIncorrectName](https://github.com/pingcap/tidb/blob/64be9ec4c0c59b7f9827b1b5b0318ade177afb9a/planner/core/preprocess.go#L1462) by moving it to some `util` package.

BTW, `[len(normalizedValue)-1] == ' '` is weird, why not just using `strings.HasSuffix(' ')`?